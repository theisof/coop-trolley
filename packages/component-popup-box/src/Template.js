import React from 'react'

import './styles.scss'

class PopupBox extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      closing: false
    }

    this.buttonClick = this.buttonClick.bind(this)
  }

  buttonClick (event) {
    if (! this.state.closing) {
      const index = event.target.dataset.index
      const action = this.props.options.actions[index]

      action && action.callback && action.callback()
      this.setState({ closing: true })

      setTimeout(() => {
        this.props.remove()
      }, 400)
    }
  }

  render () {
    const { options } = this.props

    return (
      <div className={`coop-popup-box__wrap ${this.state.closing ? 'coop-popup-box__wrap--closing' : ''}`}>
        <div className='coop-popup-box__inner'>
          <h3 className='coop-popup-box__title'>
            {options.title ? options.title : ''}
          </h3>

          <p className='coop-popup-box__text'>
            {options.text ? options.text : ''}
          </p>

          {options.actions.map((a, i) =>
            <div className={`coop-popup-box__button ${a.secondary ? 'coop-popup-box__button--secondary' : ''}`} onClick={this.buttonClick} data-index={i} key={i}>
              {a.text}
            </div>
          )}
        </div>
      </div>
    )
  }
}

export default PopupBox
