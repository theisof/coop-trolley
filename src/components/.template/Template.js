import React from 'react'

class Template extends React.Component {
  constructor(props) {
    super(props);

    this.state = {}
  }

  render() {
    return (
      <div>Hello Template</div>
    )
  }
}

export default Template