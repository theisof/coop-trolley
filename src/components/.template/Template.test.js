import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json';
import Template from './Template'

it('Template renders correctly', () => {
  const tree = shallow( <Template />)

  expect(toJSON(tree)).toMatchSnapshot()
})
