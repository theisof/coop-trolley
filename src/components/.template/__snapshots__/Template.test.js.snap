// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template renders correctly 1`] = `
<div>
  Hello Template
</div>
`;
